simurgh/migrate.py

```diff
@@ -1,18 +1,81 @@
-# migrate.py is a
+from dotenv import load_dotenv
+import os
 
-import pymssql as mssql
-import psycopg2 as pg
+import pandas as pd
+from sqlalchemy import create_engine, exc, create_engine, exc, MetaData, Table, Column, VARCHAR, Numeric, TIMESTAMP, text
+        
+load_dotenv()
 
+mssql_user = os.environ['MSSQL_USER']
+mssql_pass = os.environ['MSSQL_PASSWORD']
+mssql_host = os.environ['MSSQL_HOST']
+mssql_database = os.environ['MSSQL_DATABASE']
+
+def connect_mssql():
+    # define mssql connection object and engine
+    # https://learn.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver16
+    mssql_url_object = f"mssql+pyodbc://{mssql_user}:{mssql_pass}@{mssql_host}/{mssql_database}?driver=ODBC+Driver+17+for+SQL+Server"
+    mssql_engine = create_engine(mssql_url_object)
+    mssql_engine.connect()
+
+    return mssql_engine
+
+def connect_psql():
+    '''these are the psql creds'''
+    psql_user = "postgres"
+    psql_pass = "postgrespw"
+    psql_host = "localhost:5432"
+    psql_database = "postgres"
+
+    # define psql connection object and engine
+    psql_url_object = f"postgresql://{psql_user}:{psql_pass}@{psql_host}/{psql_database}"
+    psql_engine = create_engine(psql_url_object)
+    # define PSQL table structure
+    buildings_table_psql = Table(
```

Comment:
Let's not predispose ourselves to only ORMs, either. There's a reason why we write our SQL queries directly in the `Go` project. It might have something to do with my hellish experience with `gorm` at Cruise.

Reply:
Fair, but since this is Rony's script we should look into talking together about adjustments if necessary. At this moment though I see no problem since its readable, clean, works, and has community-backing. Especially for an early version